Update Istio version to 1.25.1 (#1365)
kyma-project/istio: docs/release-notes/1.17.0.md

##  Istio Updated to Version 1.25.1

We've updated the Istio version to 1.25.1 ([#1365](https://github.com/kyma-project/istio/pull/1365)).

> Read the [Istio 1.25.1 release announcement](https://istio.io/latest/news/releases/1.25.x/announcing-1.25.1/) and [Upgrade notes](https://istio.io/latest/news/releases/1.25.x/announcing-1.25/upgrade-notes/).

Reviewer: ```suggestion
Read the [Istio 1.25.1 release announcement](https://istio.io/latest/news/releases/1.25.x/announcing-1.25.1/) and [Istio 1.25 Upgrade Notes](https://istio.io/latest/news/releases/1.25.x/announcing-1.25/upgrade-notes/).
```

| **ENABLE\_ENHANCED\_DESTINATIONRULE\_MERGE**              | If enabled, Istio merges DestinationRules based on their **exportTo** fields. The DestinationRules are then kept as independent rules if the **exportTo** fields are not equal. |
| **PILOT\_UNIFIED\_SIDECAR\_SCOPE**                        | If true, unified SidecarScope creation is used. This is only intended as a temporary feature flag for backward compatibility.                                                   |
| **ENABLE\_DEFERRED\_STATS\_CREATION**                     | If enabled, Istio lazily initializes a subset of the stats.                                                                                                                     |
| **BYPASS\_OVERLOAD\_MANAGER\_FOR\_STATIC\_LISTENERS**     | If enabled, the overload manager is not applied to static listeners.                                                                                                            |

> This compatibility version applies the following new parameters set to `false`:

Reviewer: ```suggestion
This compatibility version applies the following new parameter set to `false`:
```

| Parameter                           | Description                                                                                                                                                                                                                |
|-------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | **PILOT\_ENABLE\_IP\_AUTOALLOCATE** | If enabled, pilot will start a controller that assigns IP addresses to ServiceEntry which do not have a user-supplied IP. This, when combined with DNS capture allows for tcp routing of traffic sent to the ServiceEntry. |

Reviewer: ```suggestion
| **PILOT\_ENABLE\_IP\_AUTOALLOCATE** | If enabled, the pilot starts a controller that assigns IP addresses to ServiceEntry that do not have a user-supplied IP. When combined with DNS capture, this allows TCP routing of traffic sent to the ServiceEntry. |
```

### Deprecation

> - `ISTIO_META_DNS_AUTO_ALLOCATE` in `proxyMetadata` has been deprecated in favor of a newer version of [DNS auto-allocation](https://istio.io/latest/docs/ops/configuration/traffic-management/dns-proxy/#address-auto-allocation)

Reviewer: ```suggestion
- `ISTIO_META_DNS_AUTO_ALLOCATE` in `proxyMetadata` has been deprecated in favor of a newer version of [DNS auto-allocation](https://istio.io/latest/docs/ops/configuration/traffic-management/dns-proxy/#address-auto-allocation).
```
